### PR TITLE
Allow only past or current dates as land purchase date

### PR DESCRIPTION
**Pull request by jakub-lotko**

### CODE REVIEW

File: src/HE.InvestmentLoans.BusinessLogic/Extensions/DateTimeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace HE.InvestmentLoans.BusinessLogic.Extensions
+{
+    public static class DateTimeExtensions
+    {
+        public static bool IsBeforeOrEqualTo(this DateTime date, DateTime otherDate) => date <= otherDate;
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/Extensions/DateTimeExtensions.cs:7]: I would be nice to have unit tests for the method

**Review by lukasz-ciura-pwc — APPROVED** [review]

File: src/HE.InvestmentLoans.BusinessLogic/Extensions/DateTimeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace HE.InvestmentLoans.BusinessLogic.Extensions
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/Extensions/DateTimeExtensions.cs:3]: Please move it to `HE.InvestmentLoans.Common.Extensions`

**Review by lukasz-ciura-pwc — APPROVED** [review]

**Review by dawid-pawulski — APPROVED** [review]

File: src/HE.InvestmentLoans.BusinessLogic/Extensions/DateTimeExtensions.cs (hunk 2 above)
Comment (jakub-lotko) [reply src/HE.InvestmentLoans.BusinessLogic/Extensions/DateTimeExtensions.cs]: File moved to HE.InvestmentLoans.Common.Extensions

File: src/HE.InvestmentLoans.BusinessLogic/Extensions/DateTimeExtensions.cs (hunk 1 above)
Comment (jakub-lotko) [reply src/HE.InvestmentLoans.BusinessLogic/Extensions/DateTimeExtensions.cs]: Tests added

File: src/HE.InvestmentLoans.BusinessLogic.Tests/Common/DateTimeExtensionTests/IsBeforeOrEqualToTests.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using FluentAssertions;
+using HE.InvestmentLoans.Common.Extensions;
+
+namespace HE.InvestmentLoans.BusinessLogic.Tests.Common.DateTimeExtensionTests
+{
+    [TestClass]
+    public class IsBeforeOrEqualToTests
+    {
+        [TestMethod]
+        public void Return_true_when_provided_date_is_before_other_date()
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic.Tests/Common/DateTimeExtensionTests/IsBeforeOrEqualToTests.cs:10]: My suggestion is to use UpperCamelCase, it feels more .NETish ^_^

**Review by lukasz-ciura-pwc — APPROVED** [review]

**Pull request merged**